src/transaction.rs

```diff
@@ -100,6 +149,76 @@ impl Transaction {
         tx.signature = Some(Signature::new(v, r, s));
         tx
     }
+
+    pub fn sender(&self) -> Result<Address, Error> {
+        if self.signature.is_none() {
+            return Ok(Address::from([0xffu8; 20]));
+        }
+        let sig = self.signature.as_ref().unwrap();
+        if sig.r == BigEndianInt::zero() && sig.s == BigEndianInt::zero() {
+            return Ok(Address::from([0xffu8; 20]));
+        } else {
+            let (vee, sighash) = if sig.v == 27u32.into() || sig.v == 28u32.into() {
```

Comment:
comments in dense code like this would be appreciated

Reply:
Updated with references to the yellow paper.